File: pec-observation-app/ios-app/Sources/PECObservation/Domain/Entities/School.swift
import Foundation

public struct School: Identifiable, Equatable, Codable {
    public let id: UUID
    public var name: String
    public var city: String
    public var district: String
    public var externalReference: String?

    public init(id: UUID = UUID(), name: String, city: String,
                district: String, externalReference: String? = nil) {
        self.id = id
        self.name = name
        self.city = city
        self.district = district
        self.externalReference = externalReference
    }
}

File: pec-observation-app/services/ms-002-school/app/models/school.py
from datetime import datetime
from sqlalchemy import Boolean, Column, DateTime, String
from pec_shared.models_base import Base, gen_uuid


class School(Base):
    __tablename__ = "schools"

    id = Column(String, primary_key=True, default=gen_uuid)
    name = Column(String, nullable=False)
    city = Column(String, nullable=False)
    district = Column(String, nullable=False)
    external_reference = Column(String, nullable=True)  # INEP code
    is_active = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
